storage/filestore_test.go

```diff
@@ -1,16 +1,148 @@
-package trustmanager
+package storage
 
 import (
-	"crypto/rand"
-	"fmt"
-	"github.com/stretchr/testify/require"
 	"io/ioutil"
 	"os"
+	"path"
 	"path/filepath"
-	"strconv"
 	"testing"
+
+	"crypto/rand"
+	"fmt"
+	"github.com/docker/notary"
+	"github.com/stretchr/testify/require"
+	"strconv"
 )
 
+const testDir = "/tmp/testFilesystemStore/"
+
+func TestNewFilesystemStore(t *testing.T) {
+	_, err := NewFilesystemStore(testDir, "metadata", "json")
+	require.Nil(t, err, "Initializing FilesystemStore returned unexpected error: %v", err)
+	defer os.RemoveAll(testDir)
+
+	info, err := os.Stat(path.Join(testDir, "metadata"))
+	require.Nil(t, err, "Error attempting to stat metadata dir: %v", err)
+	require.NotNil(t, info, "Nil FileInfo from stat on metadata dir")
+	require.True(t, 0700&info.Mode() != 0, "Metadata directory is not writable")
+}
+
+func TestSet(t *testing.T) {
+	s, err := NewFilesystemStore(testDir, "metadata", "json")
+	require.Nil(t, err, "Initializing FilesystemStore returned unexpected error: %v", err)
+	defer os.RemoveAll(testDir)
+
+	testContent := []byte("test data")
+
+	err = s.Set("testMeta", testContent)
+	require.Nil(t, err, "SetMeta returned unexpected error: %v", err)
+
+	content, err := ioutil.ReadFile(path.Join(testDir, "metadata", "testMeta.json"))
+	require.Nil(t, err, "Error reading file: %v", err)
+	require.Equal(t, testContent, content, "Content written to file was corrupted.")
+}
+
+func TestSetWithNoParentDirectory(t *testing.T) {
+	s, err := NewFilesystemStore(testDir, "metadata", "json")
+	require.Nil(t, err, "Initializing FilesystemStore returned unexpected error: %v", err)
+	defer os.RemoveAll(testDir)
+
+	testContent := []byte("test data")
+
+	err = s.Set("noexist/"+"testMeta", testContent)
+	require.Nil(t, err, "SetMeta returned unexpected error: %v", err)
+
+	content, err := ioutil.ReadFile(path.Join(testDir, "metadata", "noexist/testMeta.json"))
+	require.Nil(t, err, "Error reading file: %v", err)
+	require.Equal(t, testContent, content, "Content written to file was corrupted.")
+}
+
+// if something already existed there, remove it first and write a new file
+func TestSetRemovesExistingFileBeforeWriting(t *testing.T) {
+	s, err := NewFilesystemStore(testDir, "metadata", "json")
+	require.Nil(t, err, "Initializing FilesystemStore returned unexpected error: %v", err)
+	defer os.RemoveAll(testDir)
+
+	// make a directory where we want metadata to go
+	os.Mkdir(filepath.Join(testDir, "metadata", "root.json"), 0700)
+
+	testContent := []byte("test data")
+	err = s.Set("root", testContent)
+	require.NoError(t, err, "SetMeta returned unexpected error: %v", err)
```

Comment:
non-blocking nit: seems like `SetMeta` and `GetMeta` are still used in error messages in tests, and could be updated to the new namings